### ISSUE TITLE
使用API，ops.sort时无法跑出答案

### ISSUE THREAD

**Issue by FLoutione** [mindspore-ai/mindspore#234]
### Hardware Environment(`Ascend`/`GPU`/`CPU`):
> Uncomment only one ` /device <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> `/device gpu`</br>

### Software Environment:
- **MindSpore version (source or binary)**:2.0.0rc1.20230504
- **Python version (e.g., Python 3.7.5)**:python 3.7.13
- **OS platform and distribution (e.g., Linux Ubuntu 16.04)**:Ubuntu 20.04.4 LTS
- **GCC/Compiler version (if compiled from source)**: (Ubuntu 7.5.0-6ubuntu2) 7.5.0

## Describe the current behavior
使用这个API来对一个Tensor进行排序的时候，调试模式下转了接近10分钟没出任何结果，Tensor的shape为[30000]，dtype是Float32。
![image](https://github.com/mindspore-ai/mindspore/assets/108987302/f6b485ae-5f64-40bd-b1d5-cc5f9679a4d2)
运行这行之后，locals这里一直转圈圈
![image](https://github.com/mindspore-ai/mindspore/assets/108987302/b9cc6800-6cbf-4e58-8823-7614690a818f)
尝试接近10分钟之后没有转出来便取消了

## Describe the expected behavior
同样使用torch.sort，同样的数据，torch马上得出结果，不理解为什么mindspore这边3万条数据会一直转圈圈

## Steps to reproduce the issue
1. 
2. 
3.

## Special notes for this issue
Tensor数据中含有无穷小的值，mindspore的ops.sort不支持对含有无穷小的Tensor进行排序吗，定义的格式为-float('Inf')

**Comment by cxl-ll** [comment]
Hello, your question has been received and will be answered as soon as possible.

**Comment by 949144093** [comment]
```
import mindspore as ms                                                                                                                                           
import mindspore.ops as ops 
import pdb 

pdb.set_trace()
a = ops.randn((30000,)).astype(ms.float32)

print(a)
a[1] = -float('inf')
a[100] = -float('inf')
a[1000] = float('inf')
a[10000] = float('nan')
print(a)
b = ops.sort(a, axis=-1, descending=True)
print(b)
```
您好 我这边在GPU上用这个脚本尝试复现问题 没有复现问题 可以提供一下您能复现问题的简化脚本吗 顺便提供一下mindspore的版本 执行环境等信息

**Comment by FLoutione** [comment]
`from mindspore import Tensor, ops

if __name__ == '__main__':
# 生成包含 3 万个数据的数组，至少有两个 -inf 特殊值
    num_data = 30000
    min_inf_count = 2
    data_range = (-100, 100)  # 数据范围

    # 生成随机数据
    random_data = np.random.uniform(data_range[0], data_range[1], num_data)

    # 插入 -inf 特殊值
    inf_indices = np.random.choice(num_data, min_inf_count, replace=False)
    for idx in inf_indices:
        random_data[idx] = -float('inf')

    # 将NumPy数组转换为MindSpore张量
    input_tensor = Tensor(random_data, dtype=mindspore.float32)
    print("原始数组:", input_tensor.asnumpy())

    # 使用ops.sort进行排序，默认为升序排序
    sorted_logits, sorted_indices = ops.sort(input_tensor, axis=-1, descending=True)
    
    # 打印排序结果
    print("排序后数组:", sorted_logits.asnumpy())`
您好，这个简化代码无法跑出最终结果，在Ubuntu服务器GPU上无法跑出。但是在前面添加一句：
`# 设置运行环境
    context.set_context(mode=context.PYNATIVE_MODE, device_target="CPU")`
可以直接跑出最后答案。

MindSpore Version: 2.0.1.20230727
Python version: python 3.7.13
OS platform and distribution: Ubuntu 20.04.4 LTS
GCC/Compiler version: (Ubuntu 7.5.0-6ubuntu2) 7.5.0

**Comment by 949144093** [comment]
谢谢您的回复，问题已复现，这边会尽快处理

**Comment by 949144093** [comment]
https://e.gitee.com/mind_spore/projects/67813/bugs/list?issue=I7X5KE
